File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = ([first, second, ...rest]) => {
  return parseInt(colorCode(first).toString() + colorCode(second).toString())
};

//
// This is only a SKELETON file for the 'Resistor Color' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const colorCode = color => {
  const colors = {};
  COLORS.map((el, index) => colors[el] = index);
  return colors[color];
};

export const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
def value(colors):
    color_codes = {v: str(k) for (k, v) in enumerate(["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"])}
    return int("".join([color_codes[color] for color in colors[:2]]))
